packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -212,13 +214,13 @@ export class ZoweVsCodeExtension {
         serviceProfile: string | imperative.IProfileLoaded,
         zeRegister?: Types.IApiRegisterClient, // ZoweExplorerApiRegister
         zeProfiles?: ProfilesCache // Profiles extends ProfilesCache
-    ): Promise<void> {
+    ): Promise<boolean> {
         const cache: ProfilesCache = zeProfiles ?? ZoweVsCodeExtension.profilesCache;
-        const baseProfile = await cache.fetchBaseProfile();
+        if (typeof serviceProfile === "string") {
+            serviceProfile = await ZoweVsCodeExtension.getServiceProfileForAuthPurposes(cache, serviceProfile);
+        }
```

Comment:
Will this work as intended from a JavaScript/TypeScript perspective?
If `serviceProfile` is the `string` type, it won't update the parameter outside of the function. Maybe I'm misunderstanding the point of the assignment - if we don't use `serviceProfile` outside of the function, this is probably fine

Reply:
We were already setting `serviceProfile` this way, and AFAIK we don't use it outside of the function.

Reply:
That's fair - feel free to disregard 🙂